fastapi/routing.py

```diff
@@ -627,6 +628,16 @@ def include_router(
             responses = {}
         for route in router.routes:
             if isinstance(route, APIRoute):
+                if "GET" in route.methods and "HEAD" not in route.methods:
+                    routes_in_path = [
+                        r
+                        for r in router.routes
+                        if isinstance(r, APIRoute) and r.path == route.path
+                    ]
+                    methods_in_path = [route.methods for route in routes_in_path]
+                    all_methods = set(itertools.chain.from_iterable(methods_in_path))
+                    if "HEAD" not in all_methods:
+                        route.methods.add("HEAD")
```

Comment:
There is an edge case that might cause troubles with the current implementation:

```python

from fastapi import FastAPI
from fastapi.routing import APIRouter

router = APIRouter(prefix="/router")
router2 = APIRouter(prefix="/router")


@router.get("/")
def read_root():
    return {"Hello": "World"}

@router2.head("/")
def modified():
    print("Custom head implementation")
    return {"Hello": "World"}


app.include_router(router)
app.include_router(router2)
```

In this case, the custom endpoint will be ignored as the builtin HEAD will be appended earlier to the `routes` list at the FastAPI class